include/openssl/core_names.h

```diff
@@ -30,6 +30,59 @@ extern "C" {
 #define OSSL_PROV_PARAM_SELF_TEST_TYPE   "st-type"  /* utf8_string */
 #define OSSL_PROV_PARAM_SELF_TEST_DESC   "st-desc"  /* utf8_string */
 
+/*-
+ * Provider objects
+ *
+ * These are used when a provider wants to pass object data or an object
+ * reference back to libcrypto.  This is only useful for provider functions
+ * that take a callback to which an OSSL_PARAM array with these parameters
+ * can be passed.
+ *
+ * The object type (OSSL_OBJECT_PARAM_TYPE) is an integer that indicates
+ * the libcrypto API that can handle the data.  The known numbers are
+ * available in <openssl/object.h>.
+ * This parameter may not be present, making the object type unknown; context
+ * determines if this can be handled or not (central deserializer code will
+ * try to interpret it).
+ *
+ * The object data type (OSSL_OBJECT_PARAM_DATA_TYPE) is a string that
+ * further describes the object.  The exact interpretation of that string
+ * depends on the object type:
+ * - For OSSL_OBJECT_PKEY objects, libcrypto interprets the object data type
+ *   as a EVP_KEYMGMT algorithm name.
+ * - For objects with an unknown object type (OSSL_OBJECT_PARAM_TYPE missing
```

Comment:
not sure I understand this part starting from 'missing'..

Reply:
Maybe I should remove this comment and refer to provider-object(7)...

Reply:
It needs an update either way..